### PR TITLE
Fix sip import for distributions packaging it as sip

### LINKED ISSUE #369: PyQt6 support breaks PyQt5 on Ubuntu
I'm currently facing an issue in my CI which looks related to a recent change in pytest-qt:

The culprit is this line here: https://github.com/pytest-dev/pytest-qt/blob/master/src/pytestqt/qt_compat.py#L144 which was introduced 3 months ago.

System: Ubuntu Focal
Qt version: 5.12.8
PyQt5 version: 5.14.1

log:
```  
INTERNALERROR> Traceback (most recent call last):
18:14:56
  
18:14:56
  INTERNALERROR>   File "/usr/lib/python3/dist-packages/_pytest/main.py", line 202, in wrap_session
18:14:56
  
18:14:56
  INTERNALERROR>     config._do_configure()
18:14:56
  
18:14:56
  INTERNALERROR>   File "/usr/lib/python3/dist-packages/_pytest/config/__init__.py", line 723, in _do_configure
18:14:56
  
18:14:56
  INTERNALERROR>     self.hook.pytest_configure.call_historic(kwargs=dict(config=self))
18:14:56
  
18:14:56
  INTERNALERROR>   File "/usr/lib/python3/dist-packages/pluggy/hooks.py", line 308, in call_historic
18:14:56
  
18:14:56
  INTERNALERROR>     res = self._hookexec(self, self.get_hookimpls(), kwargs)
18:14:56
  
18:14:56
  INTERNALERROR>   File "/usr/lib/python3/dist-packages/pluggy/manager.py", line 92, in _hookexec
18:14:56
  
18:14:56
  INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
18:14:56
  
18:14:56
  INTERNALERROR>   File "/usr/lib/python3/dist-packages/pluggy/manager.py", line 83, in <lambda>
18:14:56
  
18:14:56
  INTERNALERROR>     self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
18:14:56
  
18:14:56
  INTERNALERROR>   File "/usr/lib/python3/dist-packages/pluggy/callers.py", line 208, in _multicall
18:14:56
  
18:14:56
  INTERNALERROR>     return outcome.get_result()
18:14:56
  
18:14:56
  INTERNALERROR>   File "/usr/lib/python3/dist-packages/pluggy/callers.py", line 80, in get_result
18:14:56
  
18:14:56
  INTERNALERROR>     raise ex[1].with_traceback(ex[2])
18:14:56
  
18:14:56
  INTERNALERROR>   File "/usr/lib/python3/dist-packages/pluggy/callers.py", line 187, in _multicall
18:14:56
  
18:14:56
  INTERNALERROR>     res = hook_impl.function(*args)
18:14:56
  
18:14:56
  INTERNALERROR>   File "/usr/local/lib/python3.8/dist-packages/pytestqt/plugin.py", line 203, in pytest_configure
18:14:56
  
18:14:56
  INTERNALERROR>     qt_api.set_qt_api(config.getini("qt_api"))
18:14:56
 INTERNALERROR>   File "/usr/local/lib/python3.8/dist-packages/pytestqt/qt_compat.py", line 144, in set_qt_api
18:14:56
  
18:14:56
  INTERNALERROR>     self.isdeleted = _import_module("sip").isdeleted
18:14:56
  
18:14:56
  INTERNALERROR>   File "/usr/local/lib/python3.8/dist-packages/pytestqt/qt_compat.py", line 101, in _import_module
18:14:56
  
18:14:56
  INTERNALERROR>     return getattr(m, module_name)
18:14:56
  
18:14:56
  INTERNALERROR> AttributeError: module 'PyQt5' has no attribute 'sip'
18:14:56
  
18:14:56
  [Testcase: testtest_lib] ... FAILURE!
18:14:56
  
  
```

### PR DESCRIPTION
**Pull request by The-Compiler**
Speculatively fixes #369

### CODE REVIEW

File: src/pytestqt/qt_compat.py [hunk 1]
```diff
@@ -141,7 +141,12 @@ def _import_module(module_name):
             self.isdeleted = lambda obj: not shiboken6.isValid(obj)
         else:
             assert self.is_pyqt
-            self.isdeleted = _import_module("sip").isdeleted
+            try:
+                sip = _import_module("sip")
+            except AttributeError:
+                # some distributions still package PyQt5.sip as sip
```
Comment (nicoddemus) [inline src/pytestqt/qt_compat.py:147]:
```suggestion
                # some distributions still package PyQt5.sip as sip (#396)
```

**Review by nicoddemus — DISMISSED** [review]
LGTM, it would be good if others could test it first, but seems like a no-brainer too. 👍

**Comment by machinekoder** [comment]
I can confirm the fix works on Ubuntu Focal

**Comment by nicoddemus** [comment]
From my POV we can release this as is. 👍

**Comment by The-Compiler** [comment]
I'm taking this as an approving review and force-merging then :wink: (after all, all I did in the latest revision is applying a comment change from @nicoddemus anyways)

**Comment by The-Compiler** [comment]
4.0.1 is released!

**Pull request merged**